src/components/SlidingPanel/SlidingPanel.js

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import GearButton from './GearButton';
+import SideBar from './SideBar';
+import Overlay from './Overlay';
+
+function SlidingPanel() {
+  const [isSlidingPanelOpen, setSlidingPanelVisible] = useState(false);
```

Comment:
No need to be that explicit, since it is in SlidingPanel component context SlidingPanel prefix is not needed. **isVisible** and **setIsVisible** should be ok.